docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -1525,6 +1543,11 @@ def get_person_variables(fieldslist,
   else:
     return people
 
+def set_custom_people_map( people_var_names ):
+  """Sets the map of custom people created by the developer."""
+  for var_name in people_var_names:
+    custom_values.people_plurals_map[ var_name ] = var_name
+  return custom_values.people_plurals_map
```

Comment:
I think I might be missing something, but are we ever using `custom_values.people_plurals_map` differently than `reserved_pluralizers`?  We will even want to? If it's a no to both of those feel like it makes more sense to just add the custom people directly to the `RESERVED_PLURALIZERS_MAP`, and maybe just moving that outside of `generator_constants` to something else, or just renaming `constants` to maybe `generator_reserved`? 

Over all it just feels like this PR in duplicating a lot checks between custom_values.people_plurals_map and reserved_pluralizers.

Reply:
Sorry, I meant to add these sentiments to the description. I wasn't sure what folks wanted and what direction the architecture is heading in. This was a hotfix from what I understood, so I did the quick thing that would also be very visible so that it would be easier to spot if change was desired. Among other things,  I wasn't sure if we'd want to keep these constants as constants, but wasn't going to go around mutating constants. I'm also unsure about making a whole new file for this one need, but was weighing that against other choices, like making a global variable. The list goes on.

Happy to make whatever changes folks want. If the fix is urgent and this code at least improves that sufficiently, we can always make improvements after the fact.

Reply:
Maybe worth mentioning - I also wanted to check on whether there's some point at which we want _only_ our own `RESERVED_PLURALIZERS_MAP` _without_ the custom values added in. If so, the ability to mutate the value might lead to extra work later on for someone tracking down behavior. A known constant can be relied on while a mutated value needs to be traced more thoroughly.

I wondered if the use of `RESERVED_PLURALIZERS_MAP` in `is_reserved_docx_label` and of `RESERVED_VAR_PLURALS ` in `is_reserved_label` were problematic situations, but I think not. From what I could tell, adding custom people values to `built_in_fields_used` as early as possible would just save us some work later. We might want to change the name of those functions, though. That said, I wasn't sure I hadn't missed something else.

If we do want to explore a little improvement to architecture, do you think that moving `reserved_pluralizers_map` into custom_values.py would be the right move? Creating some containing object in one of our other files? I was also wondering if we might remove `RESERVED_VAR_PLURALS` completely. It looks to me like `RESERVED_PLURALIZERS_MAP` is  an object version of the `RESERVED_VAR_PLURALS` 
list and is being used in place of the list most of the time anyway using the `.values()` method.

Reply:
Fair, if we want to hotfix this, let's get it in. But it does seem to add some cruft that we should remove later.

>  I also wanted to check on whether there's some point at which we want only our own RESERVED_PLURALIZERS_MAP without the custom values added in

I can't really think of any reason: we're advertising this as "your variables can be treated just like normal reserved vars", and since we're already doing some weird transformations behind the scenes, it would be easier for interview authors if there are fewer caveats and special cases for them to remember. And if we can't think of a specific reason, then we shouldn't; it's a good rule of thumb to avoid over engineering. 

> We might want to change the name of those functions [is_reserved_docx_label and is_reserved_label], though

IMO no, we'd just want to change the names with `constants` in them. The user defined variables are still reserved, from what I understood about the feature, this is just an easier way for them to use reserved variables without forking and editing the weaver.

> I was also wondering if we might remove RESERVED_VAR_PLURALS completely. It looks to me like RESERVED_PLURALIZERS_MAP is an object version of the RESERVED_VAR_PLURALS
list and is being used in place of the list most of the time anyway using the .values() method.

+1, I've been eyeing that lately as well.

Reply:
All that sounds good to me. Shall I implement some changes, or leave it as it is?

Reply:
Tbh I didn't realize this was just a hotfix when I commented. So I'd leave as is for now, and this can be a specific refactor ticket, I'll synthesize an issues from these comments.